packages/gatsby/src/bootstrap/load-themes/index.ts

```diff
@@ -1,29 +1,43 @@
-const { createRequireFromPath } = require(`gatsby-core-utils`)
-const path = require(`path`)
+import { createRequireFromPath } from "gatsby-core-utils"
+import * as path from "path"
 import { mergeGatsbyConfig } from "../../utils/merge-gatsby-config"
-const Promise = require(`bluebird`)
-const _ = require(`lodash`)
-const debug = require(`debug`)(`gatsby:load-themes`)
+import Promise from "bluebird"
+import _ from "lodash"
 import { preferDefault } from "../prefer-default"
 import { getConfigFile } from "../get-config-file"
-const { resolvePlugin } = require(`../load-plugins/load`)
-const reporter = require(`gatsby-cli/lib/reporter`)
+import { resolvePlugin } from "../load-plugins/load"
+import reporter from "gatsby-cli/lib/reporter"
+import Debug from "debug"
+import { GatsbyConfig, PluginRef } from "../../../index"
 
+const debug = Debug(`gatsby:load-themes`)
+
+interface IResolvedTheme {
+  themeName: string
+  themeConfig: GatsbyConfig
+  themeSpec: PluginRef
+  themeDir: string
+  parentDir: string
+  configFilePath: string
+}
 // get the gatsby-config file for a theme
 const resolveTheme = async (
-  themeSpec,
-  configFileThatDeclaredTheme,
+  themeSpec: PluginRef,
+  configFileThatDeclaredTheme: string,
   isMainConfig = false,
-  rootDir
-) => {
-  const themeName = themeSpec.resolve || themeSpec
-  let themeDir
+  rootDir: string
+): globalThis.Promise<IResolvedTheme> => {
+  const themeName =
+    typeof themeSpec === `string` ? themeSpec : themeSpec.resolve
+  let themeDir = ``
```

Comment:
Kinda modifies the runtime behavior (since `themeDir` will never be `undefined` now), but noticed it will panic if `themeDir` is falsy. So the overall behavior stays correct.

Reply:
That's fine